src/rp2_common/hardware_pio/pio.c

```diff
@@ -87,7 +88,7 @@ bool pio_can_add_program(PIO pio, const pio_program_t *program) {
 static bool _pio_can_add_program_at_offset(PIO pio, const pio_program_t *program, uint offset) {
     assert(offset < PIO_INSTRUCTION_COUNT);
     assert(offset + program->length <= PIO_INSTRUCTION_COUNT);
```

Comment:
`pio_program_t.length` is a `uint8_t`, so I think that means the `assert(offset < PIO_INSTRUCTION_COUNT);` assertion is made redundant by the `assert(offset + program->length <= PIO_INSTRUCTION_COUNT);` assertion?

Reply:
Yes this is redundant, however it is also safe, clear, and not related to -Wsign-compare, so will leave it alone in this PR

Reply:
actually this assert should fire if you pass in a value > UINT_MAX - PIO_INSTRUCTION_COUNT, whereas the second would not, so this should be left in
(e.g. if a casted -1 gets passed in by user code somehow)

Reply:
LOL, I hadn't been considering PIO programs _that_ long :wink: